### DISCUSSION TITLE
Unigine Engine support

### DISCUSSION THREAD

**Discussion by pakinmaksim** [FirstGearGames/FishNet#531] [Feature Request]
Dear FishNet Developers,

As an independent indie game developer, I have been truly impressed with the FishNet package you've developed for Unity. Its functionality has significantly streamlined networking in game development, which I've found incredibly beneficial.

I'm writing to inquire about the possibility of you creating a similar package for [Unigine](https://unigine.com/). As you might be aware, Unigine is a powerful engine that has been gaining traction in the industry but currently lacks comprehensive networking solutions like FishNet. Developing a comparable package for Unigine would not only fill this gap but also set you as the pioneers in this space.

The recent launch of the [Unigine Asset Store](https://assets.unigine.com/) presents an excellent opportunity for the introduction of such a package. Given the absence of similar tools within the Unigine community, your package has the potential to become the go-to networking solution, setting a standard for future developments.

Moreover, considering that there are no major competitors in this area for Unigine, any video tutorials and educational content created to demonstrate the usage of your package could essentially become the benchmark for networking in Unigine.

I look forward to your thoughts on this proposition.

**Comment by unity3dDFF** [comment]
As far as I know, Unigine uses c++ It's supposed to be a big job.

**Reply by pakinmaksim** [reply]
Yes, it is possible to write in C++, but you can also use C# and UnigineScript (custom programming language). There is almost no difference. As a proof, a screenshot from the store, where you can search for packages written in C#. I am currently in the process of developing on Unigine using C#. Another language wouldn't suit me. In any case, this is a huge job, since the engine has a different main cycle. I've been trying to create an analog of this package for several months now, but the code generation section involves some heavy lifting  
  
![image](https://github.com/FirstGearGames/FishNet/assets/13965436/e0b2ccf8-9014-450f-aee3-3355843ae0c3)
[LINK](https://store.unigine.com/all?sort=title&direction=ASC&source%5B%5D=csharp&page=1)

**Comment by redeyehare** [comment]
有这个时间，更应该开发一个godot的版本。Unigine Engine不支持ios，Android，web，will be meaningless。

**Comment by FirstGearGames** [comment]
There are no plans to support other engines at this time.
